apis/Google.Devtools.Aspnet/Google.Devtools.Aspnet/CloudErrorReportingExceptionLogger.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Devtools.Clouderrorreporting.V1Beta1;
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http.ExceptionHandling;
+
+namespace Google.Devtools.Aspnet
+{
+    /// <summary>
+    ///  Google Cloud Error Reporting <see cref="ExceptionLogger"/>.
+    /// </summary>
+    /// 
+    /// <example>
+    /// public static void Register(HttpConfiguration config)
+    /// {
+    ///   // Add a catch all for the uncaught exceptions.
+    ///   config.Services.Add(typeof(IExceptionLogger),
+    ///       CloudErrorReportingExceptionLogger.Create());
+    /// }
+    /// </example>
+    /// 
+    /// <remarks>
+    /// Reports unhandled exceptions to Google Cloud Error Reporting.
+    /// Docs: https://cloud.google.com/error-reporting/docs/
+    /// </remarks>
+    public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
+    {
+        // The formated Google Cloud Platform project id.
+        private readonly string _projectId;
+
+        // The service context in which this error has occurred.
+        // See: https://cloud.google.com/error-reporting/reference/rest/v1beta1/projects.events#ServiceContext
+        private readonly ServiceContext _serviceContext;
+
+        private readonly ReportErrorsServiceClient _client;
+
+        /// <summary>
+        /// Creates an instance of <see cref="CloudErrorReportingExceptionLogger"/>
+        /// </summary>
+        /// <param name="client">The Error Reporting client.</param>
+        /// <param name="projectId">The Google Cloud Platform project ID.</param>
+        /// <param name="serviceName">An identifier of the service, such as the name of the executable or job.</param>
+        /// <param name="version">Represents the source code version that the developer provided.</param> 
+        public static CloudErrorReportingExceptionLogger Create(
+            ReportErrorsServiceClient client, string projectId, string serviceName, string version)
+        {
```

Comment:
You can use an expression-bodied member here (and below), although you certainly don't have to.